tests/unit/backend/corpora/api_server/mock_auth.py

```diff
@@ -98,41 +89,24 @@ def start(self):
             params.append(self.additional_scope)
         if self.token_duration:
             params.append(str(self.token_duration))
-        self.process = subprocess.Popen(params)
+        mock_app = MockOauthApp(self.port, self.additional_scope, self.token_duration).app
+        self.process = Process(target=mock_app.run, kwargs=dict(port=self.port, debug=True, use_reloader=False))
+        self.process.start()
         # Verify that the mock oauth server is ready (accepting requests) before starting the tests.
         self.server_okay = False
         for _ in range(5):
             try:
                 response = requests.get(f"http://localhost:{self.port}/.well-known/jwks.json")
-                if response.status_code == 200:
-                    self.server_okay = True
-                    break
+                response.raise_for_status()
             except Exception:
-                pass
-
-            # wait one second and try again
-            time.sleep(1)
+                time.sleep(0.1)
```

Comment:
Just curious - why was this reduced to 0.1 seconds? Also, what happens if the server isn't ready in 0.5 seconds?

Reply:
then we will have flakey tests, but I'm confident we won't have that problem. Basically, it either starts almost immediately or doesn't start at all.